### PR TITLE
feat: add application domain on application creation and update

### PR DESCRIPTION
**Pull request by marcambier**
https://github.com/gravitee-io/issues/issues/6383

User can now set application domain when he creates a new application, or update an existing one.
In console and portal.

### CODE REVIEW

**Review by gaetanmaisse — APPROVED** [review]
🚀

File: gravitee-apim-console-webui/docs/management-application.md [hunk 1]
```diff
@@ -2,3 +2,6 @@
 
 Set general information about your application such as its name, its type (mobile/web) and description.
 You can also give access to your application to registered groups.
+
+If relevant, you can set the domain used by your application (for instance `https://my-app.com`). It will help API publishers to properly configure access controls.
+If multiple domains are concerned, use the comma as a separator.
```
Comment (plsstephenson) [inline gravitee-apim-console-webui/docs/management-application.md:7]: TIP: You can set multiple domains separated by commas.

File: gravitee-apim-console-webui/docs/management-create-application.md [hunk 2]
```diff
@@ -6,3 +6,6 @@ managed by someone else.
 
 An application is a simple entity composed of small property which are mainly defined to describe the application
 , what is does, ...
+
+If relevant, you can set the domain used by your application (for instance `https://my-app.com`). It will help API publishers to properly configure access controls.
+If multiple domains are concerned, use the comma as a separator.
```
Comment (plsstephenson) [inline gravitee-apim-console-webui/docs/management-create-application.md:11]: See above

File: gravitee-apim-console-webui/src/management/application/creation/steps/application-creation-step1.html [hunk 3]
```diff
@@ -35,12 +35,20 @@
...
               <div ng-messages="applicationForm.description.$error">
                 <div ng-message="required">Application description is required.</div>
               </div>
             </md-input-container>
+
+            <md-input-container class="md-block">
+              <label>Domain</label>
+              <input ng-model="$ctrl.parent.application.domain" type="text" name="domain" />
+              <div class="hint" ng-if="applicationForm.domain.$valid || applicationForm.domain.$pristine">
+                Which domain your application would be using, if relevant.
```
Comment (plsstephenson) [inline gravitee-apim-console-webui/src/management/application/creation/steps/application-creation-step1.html:49]: The domain your application will use, if relevant.

File: gravitee-apim-console-webui/src/management/application/details/general/application-general.html [hunk 4]
```diff
@@ -53,6 +53,14 @@ <h2>General</h2>
                 <div ng-message="required">Application description is required.</div>
               </div>
             </md-input-container>
+
+            <md-input-container>
+              <label>Domain</label>
+              <input ng-model="$ctrl.application.domain" type="text" name="domain" />
+              <div class="hint" ng-if="formApplication.domain.$valid || formApplication.domain.$pristine">
+                Which domain your application would be using, if relevant.
```
Comment (plsstephenson) [inline gravitee-apim-console-webui/src/management/application/details/general/application-general.html:61]: See above

File: gravitee-apim-portal-webui/src/assets/i18n/en.json [hunk 5]
```diff
@@ -223,6 +223,10 @@
       "label": "Describe the goal of this application",
       "title": "Description"
```
Comment (plsstephenson) [inline gravitee-apim-portal-webui/src/assets/i18n/en.json:224]: Purpose of the application

File: gravitee-apim-portal-webui/src/assets/i18n/en.json [hunk 6]
```diff
@@ -457,6 +461,10 @@
         "label": "Description",
         "placeholder": "Just describe its purpose"
```
Comment (plsstephenson) [inline gravitee-apim-portal-webui/src/assets/i18n/en.json:462]: Purpose of the application

File: gravitee-apim-portal-webui/src/assets/i18n/en.json [hunk 7]
```diff
@@ -457,6 +461,10 @@
...
       },
+      "domain": {
+        "label": "Domain your application will use",
```
Comment (plsstephenson) [inline gravitee-apim-portal-webui/src/assets/i18n/en.json:465]: Domain used by the application

File: gravitee-apim-rest-api/gravitee-apim-rest-api-model/src/main/java/io/gravitee/rest/api/model/ApplicationEntity.java [hunk 8]
```diff
@@ -41,6 +41,9 @@
     )
     private String description;
 
+    @ApiModelProperty(value = "Domain used by the application, if relevant", example = "https://my-app.com")
+    private String domain;
+
     @ApiModelProperty(value = "Application's groups. Used to add team in your App.", example = "['MY_GROUP1', 'MY_GROUP2']")
```
Comment (plsstephenson) [inline gravitee-apim-rest-api/gravitee-apim-rest-api-model/src/main/java/io/gravitee/rest/api/model/ApplicationEntity.java:47]: Application groups. Used to add teams to your application.

File: gravitee-apim-rest-api/gravitee-apim-rest-api-portal/gravitee-apim-rest-api-portal-rest/src/main/resources/openapi.yaml [hunk 9]
```diff
@@ -3702,6 +3702,9 @@ components:
         description:
           description: Description of the application.
           type: string
+        domain:
+          description: Domain used by the application.
+          type: string
         applicationType:
           description: Type of the application (Web, Mobile, ...).
```
Comment (plsstephenson) [inline gravitee-apim-rest-api/gravitee-apim-rest-api-portal/gravitee-apim-rest-api-portal-rest/src/main/resources/openapi.yaml:3709]: Application type (for example, web or mobile)

**Pull request merged**